### PR TITLE
Categorizes Late-Join Job List

### PR DESCRIPTION
**Pull request by Krausus**
Currently, when you try to join a round that's already started, you're given an ugly list of every open position, sorted only by the order that the job datums happen to be defined. You've all seen it. It's *awful.*

This PR makes it non-awful, by organizing the jobs into appropriate categories. If there are no jobs available in a given category, it won't show up. And, it makes the window *real* big - big enough to fit *all* the jobs, so you don't have to scroll. I know how much you hate scrolling.

Also, in case you're wondering why the NT Rep, Blueshield, Magistrate, and IAAs are all lumped together with the service jobs, that just happens to be the list they're in. Not my fault.

#### After ####

:cl:
tweak: The late-join job listing is now big enough to show all the jobs at once, and organizes them into labeled categories.
/:cl:

### CODE REVIEW

**Comment by TheDZD** [comment]
Looks great.

**Comment by tigercat2000** [comment]
:ok_hand:  :+1::100:

**Comment by Regen1** [comment]
That's amazing

Thank you :+1:

**Comment by Spacemanspark** [comment]
All my yes.
:+1:

**Comment by KasparoVy** [comment]
Looks really really good, awesome stuff.

File: code/modules/mob/new_player/new_player.dm [hunk 1]
```diff
@@ -415,14 +415,53 @@
...
+		"Supply" = list(jobs = list(), titles = supply_positions, color = "#ead4ae"),
+		"Support / Service" = list(jobs = list(), titles = service_positions, color = "#cccccc")
+		)
 	for(var/datum/job/job in job_master.occupations)
 		if(job && IsJobAvailable(job.title))
-			var/active = 0
+			activePlayers[job] = 0
+			var/categorized = 0
 			// Only players with the job assigned and AFK for less than 10 minutes count as active
 			for(var/mob/M in player_list) if(M.mind && M.client && M.mind.assigned_role == job.title && M.client.inactivity <= 10 * 60 * 10)
```
Comment (tigercat2000) [inline code/modules/mob/new_player/new_player.dm]: 10 * 60 * 10

:eyes:

**Comment by FalseIncarnate** [comment]
Minor thing, but would it be possible to split that into 2 columns, instead of one really long one? It's not as bad when there are only a handful of slots open, but when you have a number of vacancies across multiple departments, it can get fairly long (as your images show).

File: code/modules/mob/new_player/new_player.dm (hunk 1 above)
Comment (TheDZD) [inline code/modules/mob/new_player/new_player.dm]: The fuck oldcoders?

**Comment by Krausus** [comment]
> Minor thing, but would it be possible to split that into 2 columns

You know, you bring up a good point. There's no reason for the window to be so small...

But you're not thinking big enough. 2 columns? I'll do you one better...

**Comment by Krausus** [comment]
Bam. **Three** columns. Bigger window. Now you can see *all* the jobs.

Only downside is, most of the time, the window will probably seem a little too big...

**Comment by Davdichan** [comment]
Why hasn't this merged yet?

**Comment by Fox-McCloud** [comment]
Very nice @Krausus 

Thank you a bunch for this.

But the question is, can it make toast, too?

**Comment by Krausus** [comment]
> But the question is, can it make toast, too?

Uhh...

Yeah, sure. It can make you toast.  :neutral_face:

**Pull request merged**